### PR TITLE
Fix flashing ESP32 device and alternative method deploy on Linux

### PR DESCRIPTION
**Pull request by shodan8192**
## Description

- Match [serial-monitor-cli](https://github.com/microsoft/serial-monitor-cli) executable name for Linux and Darwin in `src/serialportctrl.ts`
- Skip addition of executable attribute for utils, when building in Azure Pipeline in `scripts/build.ps1`
- Add executable attribute for utils in `azure-pipelines.yml`
## Motivation and Context

- Fixes nanoFramework/Home#1012

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: src/serialportctrl.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class SerialPortCtrl {
     if (os.platform() === "win32") {
         fileName = "main.exe";
     } else if (os.platform() === "linux" || os.platform() === "darwin") {
-        fileName = "main";
+        fileName = "main.out";
```
Comment (josesimoes) [inline src/serialportctrl.ts:46]: Is the out extension really meant to be there? 🤔

Comment (shodan8192) [reply src/serialportctrl.ts:46]: That's the exact name of this executable in release packages (linux.zip and darwin.zip) of serial-monitor-cli, which are downloaded by gulp as postinstall task

Comment (josesimoes) [reply src/serialportctrl.ts:46]: Understood! 

No need to change anything elsewhere when call this?

Comment (shodan8192) [reply src/serialportctrl.ts:46]: No, that's the only place requiring this change

**Review by josesimoes — APPROVED** [review]
Let's go for it! Thanks for looking into this. 👍🙂

**Pull request merged**